chromium_src/chrome/browser/notifications/notification_platform_bridge_android.cc

```diff
@@ -4,9 +4,30 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "brave/build/android/jni_headers/BraveNotificationPlatformBridge_jni.h"
+#include "brave/components/brave_ads/browser/ad_notification.h"
 #include "chrome/browser/notifications/notification_platform_bridge_android.h"
 #include "chrome/android/chrome_jni_headers/NotificationPlatformBridge_jni.h"
 
+// Allows us to close a Brave ad notification across browser sessions
+#define BRAVE_MAYBE_CLOSE_AD_NOTIFICATION                                     \
+  {                                                                           \
+    const std::string brave_ads_url_prefix = brave_ads::kBraveAdsUrlPrefix;   \
```

Comment:
I think we can just call `GetDisplayed` to get the list of current notifications in brave ads and then just call it with the correct new notification id

Reply:
Not all Android versions are able to query posted notifications. `GetDisplayed` returns an empty vector on Android. That's why (in previous implementation on `tabs`) we stored them in a file.